### PR TITLE
fwd-auth: match nginx-ingress config

### PR DESCRIPTION
**Pull request by desimone**
## Summary

These changes refactor forward-auth to support `nginx-ingress` configurations. Still a WIP, but would like a review of the proposed approach. 

This PR is not ready for merge, but I would like some feedback / review prior to making any changes to the test suite. 

I've manually tested this code works for kubernetes nginx-ingress, manual nginx, and traefik. I have not tested traefik in kubernetes. 

## Related issues

- Fixes https://github.com/pomerium/pomerium/issues/1497  

**Checklist**:

- [x] add related issues
- [ ] updated unit tests (could use some peer on how best to do this). 
- [x] ready for review

### CODE REVIEW

File: examples/nginx/httpbin.conf [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+  location @authblock {
+    internal;
+    add_header Set-Cookie $auth_cookie;
+    return 302
+      https://fwdauth.localhost.pomerium.io/?uri=$scheme://$host$request_uri;
+  }
+
+  location / {
+    proxy_pass http://httpbin;
+    include /etc/nginx/auth.conf;
```
Comment (desimone) [inline examples/nginx/httpbin.conf:20]: vs
```
		location / {
			error_page 401 = @8463518dcdc3c4286bb507adad83f7ce9d0fbe96;
```

```
		location @8463518dcdc3c4286bb507adad83f7ce9d0fbe96 {
			internal;
			
			add_header Set-Cookie $auth_cookie;
			
			return 302 https://forwardauth.localhost.pomerium.io/?uri=$scheme://$host$request_uri&rd=$pass_access_scheme://$http_host$escaped_request_uri;
		}
```

File: authorize/grpc.go [hunk 2]
```diff
@@ -65,7 +77,7 @@ func (a *Authorize) Check(ctx context.Context, in *envoy_service_auth_v2.CheckRe
 	case reply.Status == http.StatusOK:
 		return a.okResponse(reply), nil
 	case reply.Status == http.StatusUnauthorized:
-		if isForwardAuth {
+		if isForwardAuth && hreq.URL.Path == "/verify" {
```
Comment (desimone) [inline authorize/grpc.go:80]: I wish there was a better way to handle nginx's "dual" endpoint forward auth configuration.

File: authorize/grpc.go [hunk 3]
```diff
@@ -39,9 +39,21 @@ func (a *Authorize) Check(ctx context.Context, in *envoy_service_auth_v2.CheckRe
 
 	state := a.state.Load()
 
-	// maybe rewrite http request for forward auth
-	isForwardAuth := a.handleForwardAuth(in)
+	// convert the incoming envoy-style http request into a go-style http request
 	hreq := getHTTPRequestFromCheckRequest(in)
+
+	isForwardAuth := a.isForwardAuth(in)
+	if isForwardAuth {
```
Comment (travisgroth) [inline authorize/grpc.go:46]: 👍

**Review by travisgroth — COMMENTED** [review]
LGTM

**Comment by desimone** [comment]
@travisgroth \ @calebdoxsey  could I grab another review  🙏

**Review by calebdoxsey — APPROVED** [review]

**Pull request merged**